src/pages/workspace/WorkspacesListPage.tsx

```diff
@@ -547,7 +632,48 @@ function WorkspacesListPage() {
 
     useHandleBackButton(onBackButtonPress);
 
-    if (isEmptyObject(workspaces)) {
+    const data = useMemo(() => {
+        const shouldShowDomainsSection = !inputValue.trim().length && domains.length;
+
+        return [
+            !workspaces.length ? [{listItemType: 'workspaces-empty-state' as const}] : [],
+            filteredWorkspaces,
+            shouldShowDomainsSection ? [{listItemType: 'domains-header' as const}, ...domains] : [],
+        ].flat();
+    }, [domains, filteredWorkspaces, workspaces.length, inputValue]);
```

Comment:
NAB: Might be good to add a comment here about what data is / what we're filtering for